### PR TITLE
CUP Unit Presets

### PR DESCRIPTION
**Pull request by Eogos**
| Q | A |
| --- | --- |
| Bug fix? | No |
| New feature? | Yes |
| Needs wipe? | No |
| Fixed issues | N/A |

### Description:

Once merged this PR will add several CUP Factions to player, enemy, resistance, and civilian sides. 

### Content:
- [x] CUP BAF (Desert)
- [x] CUP BAF (Woodland)
- [x] CUP USMC (Desert)
- [x] CUP USMC (Woodland)
- [x] CUP US Army (Desert)
- [x] CUP US Army (Woodland)
- [x] CUP CDF
- [x] CUP Army of the Czech Republic (Desert)
- [x] CUP Army of the Czech Republic (Woodland)
- [x] CUP SLA
- [x] CUP Takistani Army
- [x] CUP AFRF (MSV - EMR)
- [x] CUP AFRF (Modern MSV)
- [x] CUP Chernarussian Movement of the Red Star
- [x] CUP Takistani Locals
- [x] CUP NAPA
- [x] CUP Takistani Civilians
- [x] CUP Chernarussian Civilians

### Successfully tested on:
- [ ] Local MP
- [ ] Dedicated MP

### Compatibility checked with:
* ACE

### CODE REVIEW

**Comment by Eogos** [comment]
Gonna have to drop the CDF and Chedaki Winter versions cause apparently all the CUP winter camos for the vehicles are hidden selections instead of new separate vehicles with unique classnames. It is sad boi hours.

**Comment by veteran29** [comment]
Are you sure there no hidden classes for them? GM had something like that.

**Comment by Eogos** [comment]
It's possible, I'll go ask the guy who made them. Fairly sure they don't have them though.

**Comment by Eogos** [comment]
Nope only the selections. Unit versions of the vehicles are only available with CFP so maybe I'll do those when 0.97 drops. For now I'm sticking to just CUP tho.

**Comment by Eogos** [comment]
Nearly done now, just gotta go back and do the Modern MSV variant of the AFRF (mostly just switching camos of the infantry and moving some vehicles around) and do all the attachTo points for the logistics system for vehicles that I think logically should have it that don't already.

**Comment by Wyqer** [comment]
Sounds good.

**Comment by Eogos** [comment]
Just a heads up there are no arsenal presets to go along with any of the BLUFOR presets. I always ran unlimited arsenal with my group so I don't have any to work off of and I don't feel like cramming that in with such little time.

**Comment by Eogos** [comment]
Couldn't help myself and started working on Army of the Czech Republic BLUFOR presets as well so look forward to those as well I guess.

**Comment by Wyqer** [comment]
As soon as you're happy with this PR, remember to click on "Ready for review" at the bottom, to resolve the WIP state and we can do our reviews.

**Comment by Eogos** [comment]
Yep, of course. It should be ready by the end of today. Just working on attachTo points rn.

**Comment by Eogos** [comment]
All right I'm pretty sure that's every done now.

File: Missionframework/kp_liberation_config.sqf [hunk 1]
```diff
@@ -7,6 +7,13 @@
 KP_liberation_medical_vehicles = [
     "B_Truck_01_medical_F",
     "B_T_Truck_01_medical_F",
+	  "CUP_B_LR_Ambulance_GB_D",
+	  "CUP_B_LR_Ambulance_GB_W",
+	  "CUP_B_LR_Ambulance_CZ_W",
+	  "CUP_B_LR_Ambulance_CZ_D",
+	  "CUP_B_HMMWV_Ambulance_USA",
+	  "CUP_B_HMMWV_Ambulance_USMC",
+	  "CUP_B_BMP2_AMB_CDF",
```
Comment (Wyqer) [inline Missionframework/kp_liberation_config.sqf]: Check the indentation. Keep 4 spaces.

File: Missionframework/kp_liberation_config.sqf [hunk 2]
```diff
@@ -457,6 +482,40 @@ box_transport_config = [
...
+	["CUP_B_CH53E_USMC", -8, [0,7.566,-3], [0,5.749,-3], [0,3.847,-3], [0,2,-3], [0,0.338,-3]],
+	["CUP_B_MI6A_CDF", -8, [0,6,-0.6], [0,4.3,-0.6], [0,2.5,-0.6], [0,0.5,-0.6], [0,-1.2,-0.6], [0,-3.1,-0.6]],
+	["CUP_B_MV22_USMC", -8, [0,3.3,-1.6], [0,1,-1.6], [0,-1.3,-1.6]],
+	["CUP_B_MV22_USMC_RAMPGUN", -8, [0,3.3,-1.6], [0,1,-1.6], [0,-1.3,-1.6]],
+	["CUP_B_MV22_VIV_USMC", -8, [0,3.3,-1.6], [0,1,-1.6], [0,-1.3,-1.6]],
+	["CUP_B_CH47F_GB", -8, [0,-3.535,-1.9], [0,-1.819,-1.9], [0,0.98,-1.9], [0,3.037,-1.9]],
+	["CUP_B_CH47F_USA", -8, [0,-3.535,-1.9], [0,-1.819,-1.9], [0,0.98,-1.9], [0,3.037,-1.9]],
+	["CUP_B_Wolfhound_GMG_GB_D", -6.5, [0,-3.5,2.3]],
+	["CUP_B_Wolfhound_HMG_GB_D", -6.5, [0,-3.5,2.3]],
+	["CUP_B_Wolfhound_LMG_GB_D", -6.5, [0,-3.5,2.3]],
```
Comment (Wyqer) [inline Missionframework/kp_liberation_config.sqf]: Check the indentation. Keep 4 spaces.

File: Missionframework/presets/init_presets.sqf [hunk 3]
```diff
@@ -14,8 +14,17 @@ switch (KP_liberation_preset_blufor) do {
...
-    case 17: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\sfp_des.sqf";};
+	case 16: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_BAF_Desert.sqf";};
+	case 17: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_BAF_Woodland.sqf";};
+	case 18: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_USMC_Desert.sqf";};
+	case 19: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_USMC_Woodland.sqf";};
+	case 20: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_USA_Desert.sqf";};
+	case 21: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_USA_Woodland.sqf";};
+	case 22: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_CDF.sqf";};
+	case 23: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_ACR_Desert.sqf";};
+	case 24: {[] call compileFinal preprocessFileLineNumbers "presets\blufor\CUP_ACR_Woodland.sqf";};
```
Comment (Wyqer) [inline Missionframework/presets/init_presets.sqf]: Check the indentation. Keep 4 spaces.

File: Missionframework/presets/init_presets.sqf [hunk 4]
```diff
@@ -32,6 +41,11 @@ switch (KP_liberation_preset_opfor) do {
     case 10: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\gm_east.sqf";};
     case 11: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\gm_east_win.sqf";};
     case 12: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\unsung.sqf";};
+	case 13: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\CUP_SLA.sqf";};
+	case 14: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\CUP_TakiArmy.sqf";};
+	case 15: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\CUP_ChDKZ.sqf";};
+	case 16: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\CUP_AFRF_MSV.sqf";};
+	case 17: {[] call compileFinal preprocessFileLineNumbers "presets\opfor\CUP_AFRF_ModernMSV.sqf";};
```
Comment (Wyqer) [inline Missionframework/presets/init_presets.sqf]: Check the indentation. Keep 4 spaces.

File: Missionframework/presets/init_presets.sqf [hunk 5]
```diff
@@ -41,6 +55,8 @@ switch (KP_liberation_preset_resistance) do {
     case 3: {[] call compileFinal preprocessFileLineNumbers "presets\resistance\middle_eastern.sqf";};
     case 4: {[] call compileFinal preprocessFileLineNumbers "presets\resistance\RACS.sqf";};
     case 5: {[] call compileFinal preprocessFileLineNumbers "presets\resistance\germany.sqf";};
+	case 6: {[] call compileFinal preprocessFileLineNumbers "presets\resistance\CUP_TakiLocals.sqf";};
+	case 7: {[] call compileFinal preprocessFileLineNumbers "presets\resistance\CUP_NAPA.sqf";};
```
Comment (Wyqer) [inline Missionframework/presets/init_presets.sqf]: Check the indentation. Keep 4 spaces.

File: Missionframework/presets/init_presets.sqf [hunk 6]
```diff
@@ -50,6 +66,8 @@ switch (KP_liberation_preset_civilians) do {
     case 3: {[] call compileFinal preprocessFileLineNumbers "presets\civilians\RDSCiv.sqf";};
     case 4: {[] call compileFinal preprocessFileLineNumbers "presets\civilians\germany.sqf";};
     case 5: {[] call compileFinal preprocessFileLineNumbers "presets\civilians\unsung.sqf";};
+	case 6: {[] call compileFinal preprocessFileLineNumbers "presets\civilians\CUP_TakiCivs.sqf";};
+	case 7: {[] call compileFinal preprocessFileLineNumbers "presets\civilians\CUP_ChernoCivs.sqf";};
```
Comment (Wyqer) [inline Missionframework/presets/init_presets.sqf]: Check the indentation. Keep 4 spaces.

**Comment by Eogos** [comment]
>     * Adjust the comments in the preset files to the current ones (see the ones from the 0.96.5 branch)

Which ones are messed up still? I thought I caught all those.


>     * Adding vehicles to the resupply, rearm, refuel, etc. arrays in the config file

Damn it, I knew I forgot something.

**Comment by Eogos** [comment]
@Wyqer All requested changes should be fixed now.

**Pull request merged**